core/src/main/java/org/apache/calcite/util/format/postgresql/TimeZoneHoursFormatPattern.java

```diff
@@ -29,14 +30,57 @@
  */
 public class TimeZoneHoursFormatPattern extends StringFormatPattern {
   public TimeZoneHoursFormatPattern() {
-    super("TZH");
+    super(ChronoUnitEnum.TIMEZONE_HOURS, "TZH");
   }
 
-  @Override String dateTimeToString(ZonedDateTime dateTime, boolean haveFillMode,
+  @Override protected int parseValue(final ParsePosition inputPosition, final String input,
+      final Locale locale, final boolean haveFillMode, boolean enforceLength) throws Exception {
+
+    int inputOffset = inputPosition.getIndex();
+    String inputTrimmed = input.substring(inputOffset);
+
+    boolean isPositive = true;
+    if (inputTrimmed.charAt(0) == '-') {
+      isPositive = false;
+    } else if (inputTrimmed.charAt(0) != '+') {
+      throw new Exception();
```

Comment:
again

Reply:
Changed to a ParseException and added a message.